drizzlepac/helpers/utils.py

```diff
@@ -0,0 +1,245 @@
+import os
+import re
+import requests
+from astropy.io import fits
+
+__all__ = ['cmp_fitshdr', 'cmp_gen_hdrkeywords',
+           'word_precision_check', 'abspath',
+           'download', 'check_url']
+
+RE_URL = re.compile('\w+://\S+')
+
+default_compare = dict(
+    ignore_keywords=['DATE', 'CAL_VER', 'CAL_VCS', 'CRDS_VER', 'CRDS_CTX'],
+    keys=['primary', 'sci', 'dq'],
+    rtol=0.000001,
+)
+
+
+def cmp_fitshdr(left, right, **kwargs):
+    """Compare FITS header values using keywords
+
+    Parameters
+    ----------
+    left, right: str
+        The FITS files to compare
+
+    keys: list
+        The header keywords to compare `left` and `right`
+        (See `defaults_compare` for initial values)
+
+    rtol: float
+        The relative difference to allow when comparing two float values either
+        in header values, image arrays, or table columns.
+        (See `defaults_compare` for initial values)
+
+    no_assert: boolean
+        Return result object instead of asserting
+
+    kwargs: dict
+        Additional arguments to be passed to `FITSDiff`
+
+    Returns
+    -------
+    None
+        Assert left and right are identical
+    """
+    assert isinstance(left, str)
+    assert isinstance(right, str)
+
+    local_kwargs = dict(
+        keys=kwargs.get('keys', default_compare['keys']),
+        rtol=kwargs.get('rtol', default_compare['rtol']),
+        ignore_keywords = kwargs.get('ignore_keywords',
+                                     default_compare['ignore_keywords'])
+    )
+
+    keys = local_kwargs['keys']
+    rtol = local_kwargs['rtol']
+    ignore_keywords = local_kwargs['ignore_keywords']
+
+    assert isinstance(keys, list)
+    assert isinstance(rtol, float)
+    assert isinstance(ignore_keywords, list)
+
+    with fits.open(left) as a:
+        with fits.open(right) as b:
+            result = fits.diff.FITSDiff(fits.HDUList([a[kw] for kw in keys]),
+                                        fits.HDUList([b[kw] for kw in keys]),
+                                        ignore_keywords=ignore_keywords,
+                                        rtol=rtol,
+                                        **kwargs)
+
+    if no_assert:
+        return result
+
+    assert result.identical, result.report()
+
+
+def cmp_gen_hdrkeywords(fitsobj, base, keywords, limit=0, start=0):
```

Comment:
Again, I am not convinced that this is not something that is already in one of the classes mentioned in http://astropy.readthedocs.io/en/latest/io/fits/usage/misc.html#differs .